clientv3/concurrency/key.go

```diff
@@ -31,34 +32,34 @@ func waitDelete(ctx context.Context, client *v3.Client, key string, rev int64) e
 	for wr = range wch {
 		for _, ev := range wr.Events {
 			if ev.Type == mvccpb.DELETE {
-				return nil
+				return &wr.Header, nil
 			}
 		}
 	}
 	if err := wr.Err(); err != nil {
-		return err
+		return nil, err
 	}
 	if err := ctx.Err(); err != nil {
-		return err
+		return nil, err
 	}
-	return fmt.Errorf("lost watcher waiting for delete")
+	return nil, fmt.Errorf("lost watcher waiting for delete")
 }
 
 // waitDeletes efficiently waits until all keys matching the prefix and no greater
 // than the create revision.
-func waitDeletes(ctx context.Context, client *v3.Client, pfx string, maxCreateRev int64) error {
+func waitDeletes(ctx context.Context, client *v3.Client, pfx string, maxCreateRev int64) (*pb.ResponseHeader, error) {
 	getOpts := append(v3.WithLastCreate(), v3.WithMaxCreateRev(maxCreateRev))
 	for {
 		resp, err := client.Get(ctx, pfx, getOpts...)
 		if err != nil {
-			return err
+			return nil, err
 		}
 		if len(resp.Kvs) == 0 {
-			return nil
+			return resp.Header, nil
 		}
 		lastKey := string(resp.Kvs[0].Key)
-		if err = waitDelete(ctx, client, lastKey, resp.Header.Revision); err != nil {
-			return err
+		if _, werr := waitDelete(ctx, client, lastKey, resp.Header.Revision); werr != nil {
```

Comment:
It's the revision the `Get` sees. The create revision for `lastKey` is already in `Kvs[0]`.